llvm/docs/LangRef.rst

```diff
@@ -11061,13 +11062,16 @@ These rules are based on the assumption that no allocated object may cross
 the unsigned address space boundary, and no allocated object may be larger
 than half the pointer index type space.
 
-If the ``inbounds`` keyword is not present, the offsets are added to the
-base address with silently-wrapping two's complement arithmetic. If the
-offsets have a different width from the pointer's index type, they are
-sign-extended or truncated to the width of the pointer's index type. The result
-value of the ``getelementptr`` may be outside the object pointed to by the base
-pointer. The result value may not necessarily be used to access memory
-though, even if it happens to point into allocated storage. See the
+If the ``inbounds`` keyword is not present, the offsets are first
```

Comment:
I don't understand why this paragraph needs to be qualified by "If the ``inbounds`` keyword is not present".

Reply:
I think the idea here was that in the inbounds case, overflow can't happen anyway, so the semantics don't really matter.

I've now done a larger restructuring, and moved the discussion of the general GEP semantics at the start, before the special inbounds semantics are discussed.

I've also added a description of how the GEP offsets are calculated, because I don't think we explained that anywhere (and it's kinda important, especially the order of trunc/ext and multiplication, and which type size is used).